src/GafferUI/AnimationGadget.cpp

```diff
@@ -724,6 +930,8 @@ bool AnimationGadget::buttonRelease( GadgetPtr gadget, const ButtonEvent &event
 
 		if( controlHeld ) // insert a keyframe
 		{
+			m_selectedKeys->clear();
+
 			insertKeyframe( curvePlug.get(), i.x );
```

Comment:
It'd be useful to select the newly inserted key here, particularly for when the tangents and editor widgets land.